Documentation/ceph-nfs-crd.md

```diff
@@ -72,55 +82,395 @@ spec:
+enabled.
+
+To enable exports, we are going to modify the Ceph RADOS object (stored in Ceph) that defines the
+configuration shared by all NFS daemons.
+
+Please note that `<pool>` and `<namespace>` will continue to refer to the configured `rados` spec's
+`pool` and `namespace` for a particular CephNFS cluster.
+
+List the shared configuration objects in a Ceph pool with this command from the Ceph toolbox.
+```console
+rados --pool <pool> --namespace <namespace> ls
+```
+
+The output may look something like below after you have created two exports. Here we have used the
+`my-nfs` example CephNFS.
+```
+conf-nfs.my-nfs
+export-1
+export-2
+grace
+rec-0000000000000002:my-nfs.a
 ```
 
-[For multiple NFS-GANESHA cluster](https://docs.ceph.com/en/latest/mgr/dashboard/#support-for-multiple-nfs-ganesha-clusters)
+The configuration of NFS daemons, and enabling exports, is controlled by the `conf-nfs.my-nfs`
+object in this example. The object name follows the `conf-nfs.<cephnfs-name>` pattern.
 
+Get the contents of the config file, which may be empty as in this example.
 ```console
-ceph dashboard set-ganesha-clusters-rados-pool-namespace <cluster_id>:<pool_name>[/<namespace>](,<cluster_id>:<pool_name>[/<namespace>])*
+rados --pool <pool> --namespace <namespace> get conf-nfs.my-nfs my-nfs.conf
+cat my-nfs.conf
 ```
 
-## NFS Settings
+Modify the `my-nfs.conf` file above to add URLs for enabling exports.
+```
+%url "rados://<pool>/<namespace>/export-1"
+%url "rados://<pool>/<namespace>/export-2"
+```
 
-### RADOS Settings
+Then write the modified file to the RADOS config object.
+```console
+rados --pool <pool> --namespace <namespace> put conf-nfs.my-nfs my-nfs.conf
+```
+
+Verify the changes are saved by getting the config again, just as before.
+```console
+rados --pool <pool> --namespace <namespace> get conf-nfs.my-nfs my-nfs.conf
+cat my-nfs.conf
+```
+```
+%url "rados://<pool>/<namespace>/export-1"
+%url "rados://<pool>/<namespace>/export-2"
+```
+
+### For Ceph v16 or newer
+For Ceph v16.2.0 through v16.2.6, exports cannot be managed through the Ceph dashboard, and
+newly-created Ceph command line tools are lacking. We highly recommend using Ceph v16.2.7 or higher
+with NFS, which fixes bugs and streamlines export management, allowing exports to be created via the
+Ceph Dashboard and the Ceph CLI. With v16.2.7 or higher, the Ceph dashboard and Ceph CLI will be
+able to manage the same NFS exports interchangeably as desired.
+
+#### Using the Ceph Dashboard
+Exports can be created via the
+[Ceph dashboard](https://docs.ceph.com/en/latest/mgr/dashboard/#nfs-ganesha-management) for Ceph v16
+as well. To enable and use the Ceph dashboard in Rook, see [here](ceph-dashboard.md).
+
+#### Using the Ceph CLI
+The Ceph CLI can be used from the Rook toolbox pod to create and manage NFS exports. To do so, first
+ensure the necessary Ceph mgr modules are enabled and that the Ceph orchestrator backend is set to
+Rook.
+```console
+ceph mgr module enable rook
```

Comment:
Shouldn't we recommend setting the enabled module in the mgr spec instead of jumping in the toolbox? Also I think the porch rook backend might be enabled already...

Reply:
Yes and I don't feel really strong about it too, also later on you recommend disabling the module so it's probably not worth editing the CR for this :).